### PR TITLE
Csi 1219 fix initial device setup issues

### PR DESCRIPTION
**Pull request by HarishRocks**

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1219 Fix initial device setup issues](https://app.clickup.com/t/37308523/CSI-1219)

File: cysync/src/store/hooks/flows/useCardAuth.ts [hunk 1]
```diff
@@ -109,6 +112,15 @@ export const useCardAuth: UseCardAuth = isInitial => {
...
     deviceState
   } = useConnection();
+  const { connected: internetConnected } = useNetwork();
+
+  useEffect(() => {
+    setEnableRetry(!!deviceConnection && internetConnected);
+  }, [deviceConnection, internetConnected]);
+
+  useEffect(() => {
+    if (errorObj.isSet && internetConnected) setShowRetry(true);
```
Comment (irshadCypherock) [inline cysync/src/store/hooks/flows/useCardAuth.ts:122]: This should also depend on errorObj, right?

File: cysync/src/errors/error.ts [hunk 2]
```diff
@@ -33,6 +33,15 @@ class DisplayError {
...
   }
+  /**
+   * Errors are structure in a way that the second sentence of the error
+   * message can be substituted for the action message.
+   * Mostly needed on Tooltips
+   * @returns action message
+   */
+  public getActionMessage() {
+    return this.message.split('.')[1];
+  }
```
Comment (irshadCypherock) [inline cysync/src/errors/error.ts:44]: Shouldn't we have a length check here? And also a warning log and default action message if the length is not correct?

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: cysync/src/store/hooks/flows/useCardAuth.ts (hunk 1 above)
Comment (HarishRocks) [reply cysync/src/store/hooks/flows/useCardAuth.ts:122]: Yes yes, adding it to the dependency list

File: cysync/src/errors/error.ts (hunk 2 above)
Comment (HarishRocks) [reply cysync/src/errors/error.ts:44]: May I know why do you feel we would need a length check here?

Comment (HarishRocks) [reply cysync/src/errors/error.ts:44]: 1. Need a default action message
2. length check after splitting
3. log if there is no second sentence

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**